### ISSUE TITLE
Efficient Matrix-Matrix Products

### ISSUE THREAD

**Issue by tbetcke** [UCL-ARC/householder#2]
Any efficient linear algebra library requires fast matrix-matrix products. The Rust ndarray project uses the matrixmultiply library https://github.com/bluss/matrixmultiply, which uses the same ideas on which the BLIS project is based. This seems to be a very good start.

In this issue we want to track the use of matrixmultiply and its performance compared to Openblas on different architectures.

**Comment by jedbrown** [comment]
Is this issue complete in the sense that householder uses matrixmultiply?

**Comment by tbetcke** [comment]
I leave it open on purpose because of performance evaluation. I'd like to have good comparison metrics against Openblas on different architectures.

**Comment by jedbrown** [comment]
Yep, or use the BLIS benchmarking infrastructure to test `matrixmultiply`. https://github.com/flame/blis/blob/master/docs/Performance.md

Or this could be done entirely in Rust, sweeping over problem sizes for each factorization and scalar, but I'd encourage plotting it normalized to match the BLIS figures.
